SitumWayfinding/Classes/Positioning/PositioningViewController.swift

```diff
@@ -1308,4 +1318,66 @@ extension PositioningViewController {
     }
 }
 
-
+extension PositioningViewController {
+    func initializeChangeOfFloorMarker() {
+        let icon = self.scaledImage(
+            image: UIImage(named: "change_floor")!,
+            scaledToSize: CGSize(width: 40.0, height: 40.0)
+        )
+        let markerView = UIImageView(image: icon)
+        self.changeOfFloorMarker.iconView = markerView
+        self.changeOfFloorMarker.position = CLLocationCoordinate2D(latitude: 0, longitude: 0)
+        resetChangeOfFloorMarker()
+    }
+    
+    func showChangeOfFloorMarker(position: CLLocationCoordinate2D) {
+        self.changeOfFloorMarker.position = position
+        self.changeOfFloorMarker.map = self.mapView
+    }
+    
+    func scaledImage(image:UIImage, scaledToSize newSize:CGSize) -> UIImage{
+        UIGraphicsBeginImageContextWithOptions(newSize, false, 0.0)
+        image.draw(in: CGRect(x: 0, y: 0, width: newSize.width, height: newSize.height))
+        let newImage:UIImage = UIGraphicsGetImageFromCurrentImageContext()!
+        UIGraphicsEndImageContext()
+        return newImage
+    }
+    
+    func updateChangeOfFloorMarkerForSelectedFloor(segment: SITRouteSegment, selectedFloor: SITFloor) {
+        if isDrawChangeOfFloorMarker(segment: segment, selectedFloor: selectedFloor) {
+            self.drawChangeOfFloorMarker(segment: segment)
+        }
+                
+        if self.lastSelectedMarker?.floorIdentifier == selectedFloor.identifier  {
+            resetChangeOfFloorMarker()
+        }
+    }
+    
+    func drawChangeOfFloorMarker(segment: SITRouteSegment) {
+        let last = segment.points.last
+        if let coordinates = last?.coordinate() {
+            let position = CLLocationCoordinate2D(
+                latitude: coordinates.latitude,
+                longitude: coordinates.longitude
+            )
+            self.showChangeOfFloorMarker(position: position)
+        }
+    }
+    
+    func isSelectedChangeofFloorMarker(segment: SITRouteSegment, selectedFloor: SITFloor) -> Bool {
```

Comment:
Aquí o que fas é comprobar se o segmento que lle pasas está no floor seleccionado, penso que o nome da función non describe o que fai. Esta función pode ser útil para outras cousas que non sexan pintar o marker de cambio de planta e polo tanto non ten sentido metelo no nome da función.
Non ten máis sentido chamarlle a funcion isSegment(SITRouteSegment, inFloor:SITFloor)